#!/usr/bin/env python2
PACKAGE = "image_denoising"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("num_frames", int_t, 0,
        "how many frames to use for denoising", 1, 1, 5)

gen.add("template_window_size", int_t, 0,
        "template window size for denoising", 7, 3, 9)

gen.add("search_window_size", int_t, 0,
        "search window size for denoising", 21, 7, 31)

gen.add("h", double_t, 0,
        "h parameter for denoising", 20.0, 1, 100.0)

exit(gen.generate(PACKAGE, "image_denoising", "NLMeansDyn"))
